### PR TITLE
feat: add support for saving safetensors in the `save_weights`

### PR DESCRIPTION
**Pull request by mzbac**
## Proposed changes

Close https://github.com/ml-explore/mlx/issues/486

## Checklist

Put an `x` in the boxes that apply.

- [x] I have read the [CONTRIBUTING](https://github.com/ml-explore/mlx/blob/main/CONTRIBUTING.md) document
- [x] I have run `pre-commit run --all-files` to format my code / installed pre-commit prior to committing changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have updated the necessary documentation (if needed)

### CODE REVIEW

File: python/mlx/nn/layers/base.py [hunk 1]
```diff
@@ -166,9 +166,20 @@ def load_weights(
 
     def save_weights(self, file: str):
         """
-        Save the model's weights to a ``.npz`` file.
+        Save the model's weights to a file. The saving method is determined by the file extension:
+        - `.npz` extension will use mx.savez
+        - `.safetensors` extension will use mx.save_safetensors
```
Comment (awni) [inline python/mlx/nn/layers/base.py:171]:
```suggestion
        - ``.npz`` extension will use :func:`mx.savez`
        - ``.safetensors`` extension will use :func:`mx.save_safetensors`
```

**Review by awni — COMMENTED** [review]
Beautiful, thanks! 

I left a minor comment for the docstring.

Also would you mind updating the docstring for `load_weights` to mention it supports `.safetensors` as well?

**Comment by mzbac** [comment]
> Beautiful, thanks!
> 
> I left a minor comment for the docstring.
> 
> Also would you mind updating the docstring for `load_weights` to mention it supports `.safetensors` as well?

Yeah, I will update it now.

**Review by awni — APPROVED** [review]
👍 thanks!

**Pull request merged**